go/vt/vttablet/tabletmanager/rpc_external_reparent.go

```diff
@@ -106,7 +107,7 @@ func (agent *ActionAgent) TabletExternallyReparented(ctx context.Context, extern
 		// tablet record. The actual record in topo will be updated later.
 		newTablet := proto.Clone(tablet).(*topodatapb.Tablet)
 		newTablet.Type = topodatapb.TabletType_MASTER
-
+		newTablet.MasterTermStartTime = logutil.TimeToProto(agent.masterTermStartTime())
```

Comment:
I don't think setting the field here carries over to the finalize stage, where we directly call UpdateTabletFields. Perhaps in general we should try to consolidate any calls to UpdateTabletFields to be done via `topotools.ChangeType` as much as possible - whenever the update is about changing type.

Reply:
I tried to change all calls to UpdateTabletFields in this file to ChangeType. There is one call in particular that started failing a test. We need to revisit that.

Reply:
I'm leaving them as UpdateTabletFields for reasons that are documented in code.